### PR TITLE
Add Windows builds on Jenkins

### PR DESCRIPTION
**Pull request by lucianomartin**
Closes https://xmosjira.atlassian.net/browse/AP-144
Closes  https://xmosjira.atlassian.net/browse/AP-146

### CODE REVIEW

**Comment by lucianomartin** [comment]
This PR is not ready.

File: Jenkinsfile [hunk 1]
```diff
@@ -96,6 +96,40 @@ pipeline {
...
+          stages {
+            stage('Build') {
+              steps {
+                runningOn(env.NODE_NAME)
+                dir('lib_xcore_math') {
+                  checkout scm
+                  // fetch submodules
+                  bat 'git submodule update --init --recursive --jobs 4'
+                  withTools(params.TOOLS_VERSION) {
+                    withVS {
```
Comment (uvvpavel) [inline Jenkinsfile:112]: not sure why you need `withVS` here as it doesn't build `x86`

**Review by uvvpavel — APPROVED** [review]
Left a few comments, otherwise looks good.

File: Jenkinsfile [hunk 2]
```diff
@@ -96,6 +96,40 @@ pipeline {
...
+                      // xs3a build
+                      bat 'cmake -B build_xs3a -DXMATH_SMOKE_TEST=${params.XMATH_SMOKE_TEST} --toolchain=etc/xmos_cmake_toolchain/xs3a.cmake -G"Ninja"'
+                      bat 'ninja -C build_xs3a -j4'
```
[the remark this answers is not in the document]
Comment (lucianomartin) [reply Jenkinsfile]: Good point, I remove that option

File: Jenkinsfile (hunk 1 above)
Comment (lucianomartin) [reply Jenkinsfile]: I don't know either, but if I remove that command I get the error below. @BrennanGit, is this behaviour expected?

```
[2023-09-28T09:32:34.311Z] C:\jenkins\workspace\XMOS_lib_xcore_math_PR-152\lib_xcore_math>cmake -B build_xs3a -DXMATH_SMOKE_TEST=${params.XMATH_SMOKE_TEST} --toolchain=etc/xmos_cmake_toolchain/xs3a.cmake -G"Ninja" 
[2023-09-28T09:32:36.825Z] -- Found Python3: C:/Python310/python3.exe (found version "3.10.7") found components: Interpreter 
[2023-09-28T09:32:36.825Z] CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
[2023-09-28T09:32:36.825Z] -- Configuring incomplete, errors occurred!
[2023-09-28T09:32:36.825Z] See also "C:/jenkins/workspace/XMOS_lib_xcore_math_PR-152/lib_xcore_math/build_xs3a/CMakeFiles/CMakeOutput.log".
script returned exit code 1
```

File: README.rst [hunk 3]
```diff
@@ -30,11 +30,23 @@ Requirements
 ------------
 
 * xTIMEcomposer tools version 15.0.6 or later
-* CMake (for building the library and unit tests)
-* GNU Make (for building the libary and unit tests)
+* CMake (for building the library and unit tests, version 3.14 or later)
```
[the remark this answers is not in the document]
Comment (lucianomartin) [reply README.rst:33]: I wouldn't change this sentence in this PR, if you think this is a problem we can create an issue. I agree it is confusing.

File: Jenkinsfile (hunk 1 above)
Comment (BrennanGit) [reply Jenkinsfile]: I had to check but it looks like ninja is installed on windows as part of the visual studio build tools.

**Pull request merged**